java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -0,0 +1,2943 @@
+   * Note that all columns must have the same type.
+   */
+  public static ColumnVector concatenate(ColumnVector... columns) {
+    if (columns.length < 2) {
+      throw new IllegalArgumentException("Concatenate requires 2 or more columns");
+    }
+    long[] columnHandles = new long[columns.length];
+    for (int i = 0; i < columns.length; ++i) {
+      columnHandles[i] = columns[i].getNativeView();
+    }
+    return new ColumnVector(concatenate(columnHandles));
+  }
+
+  /**
+   * Create a new vector of "normalized" values, where:
+   *  1. All representations of NaN (and -NaN) are replaced with the normalized NaN value
+   *  2. All elements equivalent to 0.0 (including +0.0 and -0.0) are replaced with +0.0.
+   *  3. All elements that are not equivalent to NaN or 0.0 remain unchanged.
+   *
+   * The documentation for {@link Double#longBitsToDouble(long)}
+   * describes how equivalent values of NaN/-NaN might have different bitwise representations.
+   *
+   * This method may be used to compare different bitwise values of 0.0 or NaN as logically
+   * equivalent. For instance, if these values appear in a groupby key column, without normalization
+   * 0.0 and -0.0 would be erroneously treated as distinct groups, as will each representation of NaN.
+   *
+   * @return A new ColumnVector with all elements equivalent to NaN/0.0 replaced with a normalized equivalent.
+   */
+  public final ColumnVector normalizeNANsAndZeros() {
+    return new ColumnVector(normalizeNANsAndZeros(getNativeView()));
+  }
+
+  /**
+   * Create a deep copy of the column while replacing the null mask. The resultant null mask is the
+   * bitwise merge of null masks in the columns given as arguments.
+   *
+   * @param mergeOp binary operator, currently only BITWISE_AND is supported.
+   * @param columns array of columns whose null masks are merged, must have identical number of rows.
+   * @return the new ColumnVector with merged null mask.
+   */
+  public final ColumnVector mergeAndSetValidity(BinaryOp mergeOp, ColumnView... columns) {
+    assert mergeOp == BinaryOp.BITWISE_AND : "Only BITWISE_AND supported right now";
+    long[] columnViews = new long[columns.length];
+    long size = getRowCount();
+
+    for(int i = 0; i < columns.length; i++) {
+      assert columns[i] != null : "Column vectors passed may not be null";
+      assert columns[i].getRowCount() == size : "Row count mismatch, all columns must be the same size";
+      columnViews[i] = columns[i].getNativeView();
+    }
+
+    return new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
+  }
+
+  /**
+   * Create a new vector containing the MD5 hash of each row in the table.
+   *
+   * @param columns array of columns to hash, must have identical number of rows.
+   * @return the new ColumnVector of 32 character hex strings representing each row's hash value.
+   */
+  public static ColumnVector md5Hash(ColumnView... columns) {
```

Comment:
Why do we have two copies of this?  One in ColumnVector and one in ColumnView.  static factory methods that return a ColumnVector feel to me like they belong in ColumnVector, not ColumnView.

Reply:
fixed.